synergy-design-system/synergy-design-system
# docs: 📚 Create showcase for form field usage in Storybook

# Pull Request

## 📖 Description

This PR adds the first version of a static form element, as seen on the corresponding Figma branch. It is intended to show how to build simple forms. The provided markup will be used for the framework demos to provide a coherent form demo for all supported frameworks.

### 🎫 Issues

Closes #124.

## 👩‍💻 Reviewer Notes

I have created two samples for the form, one is done in `mdx` and one in storybooks story format. Both have a set of advantages and drawbacks which should be discussed. I would opt for the regular story approach.

### MDX:

#### Advantages:

- You write (nearly) plain HTML in a single file, using react syntax.
- You can write your JS functionality inline

#### Disadvantages:

- Newlines will automatically insert a `<p />`, which is unwanted and leads to problems during formatting the code.
- There are no screenshots here.
- No possibilities to see the code in Storybook.
- There will be no screenshots captured currently (this may be configurable, I did not have a look at it).
- Code Formatting, AutoComplete and Linting not possible.

---

### CSF:

#### Advantages:

- Code Preview available via `Docs`.
- Formatting is up to you completely, no messy custom tags inserted.
- Proper syntax highlighting and type checking.

#### Disadvantages:

- Displays unneeded information (e.g. argTypes in Docs page)
- Unable to hide the "default" export from sidebar, which leads to duplicated content (Docs and Default Story)

---

Also, as this is the recommended base we want others to adhere to, we should double check accessibility and formatting issues (e.g. css adjustments). Would look forward to get some feedback, especially from @mariohamann.

I have currently opted to add the text in a static manner, this should be done via docs tokens (and maybe also exchanged for english when we are at it) later on. This can be done when @LeaLeRose starts working on the Figma template.


## 📑 Test Plan

- Check out the branch
- Issue `pnpm i -r && pnpm build`
- Start up Storybook in `packages/docs`
- Inspect the two versions at the new `Templates` navigation at the bottom.

## ✅ DoD
